plugins/contactpressuresensorsarray/src/ContactPressureSensorsArrayDriver.cpp

```diff
@@ -0,0 +1,516 @@
+    this->m_sensor->SetActive(true);
+  
+    return true;
+}
+
+bool GazeboYarpContactPressureSensorsArrayDriver::close()
+{
+    yarp::os::LockGuard guard(m_dataMutex);
+    this->m_updateConnection.reset();
+    return true;
+}
+
+bool GazeboYarpContactPressureSensorsArrayDriver::initLinkAssociatedToContactSensor(yarp::os::Property &pluginParameters)
+{
+    m_linkAssociateToSensor = pluginParameters.find("linkName").asString();
+  
+    const gazebo::physics::Link_V &gazeboModelLinks = m_robot->GetLinks();
+    std::string linkNameScopedEnding = "::" + m_linkAssociateToSensor;
+    bool linkFound = false;
+    for (size_t gazeboLink = 0; gazeboLink < gazeboModelLinks.size(); gazeboLink++)
+    {  
+        std::string gazeboLinkName = gazeboModelLinks[gazeboLink]->GetScopedName();
+        if (GazeboYarpPlugins::hasEnding(gazeboLinkName, linkNameScopedEnding))
+        {
+            linkFound = true;
+            m_sensorLink = boost::get_pointer(gazeboModelLinks[gazeboLink]);
+            break;
+        }
+    }
+  
+    if (!linkFound)
+    {
+        yError() << "GazeboYarpContactPressureSensorsArrayDriver: initContactSensor(): Link associated to sensor not found";
+        return false;
+    }
+    
+    return true;
+}
+
+bool GazeboYarpContactPressureSensorsArrayDriver::initContactSensor()
+{
+    // Choosing to get the sensor information from the sdf::Element of the link rather than gazebo::LinkPtr
+    // because Gazebo has a division between physics engine and sensors engine handled in
+    // separate threads. To avoid any runtime discrepancy, the decision to get info from sdf is made.
+    // Also the Link Element might have multiple sensors, we need specifically the contact type sensor
+    // Assumes there is only one contact collision element per link
+    size_t linkSensorCount = m_sensorLink->GetSensorCount();
+    sdf::ElementPtr sensorsdf;
+    bool contactSensorFound = false;
+    for (size_t sensor = 0; sensor < linkSensorCount; sensor++)
+    {
+        sensorsdf = (m_sensorLink->GetSDF()).get()->GetElement("sensor");
+        if (sensorsdf.get()->GetAttribute("type")->GetAsString() == "contact")
+        {
+            break;
+        }
+    }
+  
+    std::string sensorName = sensorsdf.get()->GetAttribute("name")->GetAsString();
+    // For some reason scoped name is not necessary force
+    // instantiating the contact sensor, uncomment the block if necessary
+    // Get collision name 
+    sdf::ElementPtr collision = (sensorsdf.get()->GetElement("contact")).get()->GetElement("collision"); 
+    std::string collisionName = collision.get()->GetValue()->GetAsString();
+    /* std::string tmp = m_sensorLink->GetName() + "::" + collisionName + "::" + sensorName;
+     * sensorName.clear();
+     * sensorName = tmp;
+     */
+  
+  
+    this->m_linkCollisionName = m_robot->GetName() + "::" + m_sensorLink->GetName() + "::" + collisionName;
```

Comment:
It would be worth to check if this works by trying it on a nested model, such as "iCub Fixed".

Reply:
I noticed in iCub Fixed... the collision name is set {link_name}_collision. In this case the collision name is actually set to {link_name}_fixed_joint_lump_{link_name}_collision_collision, according to the observations made in http://answers.gazebosim.org/question/15095/gazebo-contact-sensor-added-to-a-robot-link-in-urdf-file-is-not-working-as-expected/

This is not handled by the plugin. However, need to test the plugin for heavily nested models.

Reply:
I think we have implemented only a corner case. What to do with this kind of generic handling ?